packages/runtime/src/runtime/hooks/useEnsembleAction.tsx

```diff
@@ -190,6 +194,53 @@ export const useInvokeApi: EnsembleActionHook<InvokeAPIAction> = (action) => {
   return invokeApi;
 };
 
+const noneStyleOption = {
+  backgroundColor: "transparent",
+  showShadow: false,
+};
+
+export const useShowDialog: EnsembleActionHook<ShowDialogAction> = (
+  action?: ShowDialogAction,
+) => {
+  const { openModal } = useContext(ModalContext) || {};
```

Comment:
How does this behave with dialog and modal screen?

Reply:
if dialog is opened from within `navigateModalScreen`, it closes the modalScreen beneath and then open dialog. I think this is not correct way :(

Reply:
Yeah we will need to support opening a dialog on top of a modal screen